devito/ir/cgen/printer.py

```diff
@@ -370,7 +370,7 @@ def _print_UnaryOp(self, expr, op=None, parenthesize=False):
 
     def _print_Cast(self, expr):
         cast = f'({self._print(expr._C_ctype)}{self._print(expr.stars)})'
-        return self._print_UnaryOp(expr, op=cast)
+        return self._print_UnaryOp(expr, op=cast, parenthesize=True)
```

Comment:
nitpicking...

so this is to systematically generate something along the lines of ` ... = (float*) (a);` is that correct?

if so, we could maybe do instead

```
... , parenthesize=not expr.is_Atom)
```